### PR TITLE
Draft of the fulltext for the segment for product people

### PR DESCRIPTION
**Pull request by MaineC**
This builds on the detailed outline developed in #508 

It provides a fulltext version of a learning path segment for people experienced with product development. It tries to focus on the contributions that people who are not writing source code themselves can have on InnerSource projects.

### CODE REVIEW

**Comment by rrrutledge** [comment]
This work is great, @MaineC ❗  I will work to review it sometime in the upcoming week.

File: product/01-introduction.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Welcome to the learning path section for product managers.
+After watching this segment, you will have a better understanding of how InnerSource relates to Agile development best practices.
+While we all strive for autonomous teams that can move in an independent way, it is clear that in an complex, interconnected world some dependencies cannot be avoided.
+InnerSource provides an alternative to "wait it out", "build workarounds" and "escalate": Teams that need modifications in their dependencies can themselves offer a helping hand.
```
Comment (rrrutledge) [inline product/01-introduction.md:4]:
```suggestion
InnerSource [provides an alternative](https://innersourcecommons.org/learn/learning-path/introduction/02/) to "wait it out", "build workarounds" and "escalate": Teams that need modifications in their dependencies can themselves offer a helping hand.
```

File: product/01-introduction.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+InnerSource facilitates cross team collaboration.
+Through it's focus on written communication, InnerSource values are particularly helpful for teams moving to a remote first mode.
```
Comment (rrrutledge) [inline product/01-introduction.md:6]:
```suggestion
Through it's focus on written communication, InnerSource values are particularly helpful for teams moving to a remote-first mode.
```

File: product/01-introduction.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+In this Learning Path segment, you will learn where Agile development and InnerSource use similar terminology and even technology - but differ substantially in the details.
```
Comment (rrrutledge) [inline product/01-introduction.md:8]:
```suggestion
In this Learning Path section, you will learn where Agile development and InnerSource use similar terminology and even technology - but differ substantially in the details.
```

Use "Section" (not "Segment") here.  See [Content Organization](https://github.com/InnerSourceCommons/InnerSourceLearningPath/#content-organization).

File: product/01-introduction.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+Instead of running into common misunderstandings you will be better equipped to spot differences in culture but also in purpose of tools used.
```
Comment (rrrutledge) [inline product/01-introduction.md:9]:
```suggestion
Instead of running into common misunderstandings, you will be better equipped to spot differences in culture but also in purpose of tools used.
```

Comment (rrrutledge) [inline product/01-introduction.md:9]:
```suggestion
Instead of running into common misunderstandings you will be better equipped to spot differences in culture and also in purpose of tools used.
```

I've having trouble with this sentence.  How are `common misunderstanding`, `differences in culture`, and `purpose of tools` related?

File: product/01-introduction.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+In the fourth part of the segment, you will learn more about the impact of InnerSource on capacity planning: Also with InnerSource there is no free lunch - host teams will have to be aware of mentoring efforts to get contributors up to speed and mentoring them through at least the first few of their contributions.
```
Comment (rrrutledge) [inline product/01-introduction.md:11]:
```suggestion
Later in the section, you will learn more about the impact of InnerSource on capacity planning: Also with InnerSource there is no free lunch - host teams will have to be aware of mentoring efforts to get contributors up to speed and mentoring them through at least the first few of their contributions.
```

Avoid referencing specific placings of segments (in case they change).

Comment (rrrutledge) [inline product/01-introduction.md:11]:
```suggestion
In the fourth part of the segment, you will learn more about the impact of InnerSource on capacity planning.  Host teams will have to plan for mentoring efforts to get contributors up to speed through at least the first few of their contributions.
```

Shortened and used active words.

File: product/02-innersource-and-agile.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+InnerSource helps your team deliver value and maintain autonomy especially in a highly interconnected world.
+
+
+## Autonomous teams in an interconnected world
+
+Organisations try to deliver value to customers quickly.
+One common cause for delays are dependencies in the delivery process.
+As a result many organisations have moved to cross functional teams where communication with customers, design, implementation, testing and deployment/ operations are all in one team thus eliminating the need for costly handovers.
+However in order to achieve high performance teams try to also eliminate waste and re-use existing components.
+From a team perspective each re-used component however adds another link, another dependency outside of the control of that team.
```
Comment (rrrutledge) [inline product/02-innersource-and-agile.md:14]:
```suggestion
From a team perspective each re-used component adds another dependency outside of the control of that team.
```
Simplify

File: product/02-innersource-and-agile.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+Often the result are lengthy roadmap discussions to get everyone on the same page and globally optimise prioritization so that changes that are needed in central components can be implemented in an orderly fashion.
```
Comment (rrrutledge) [inline product/02-innersource-and-agile.md:15]:
```suggestion
Often the results are lengthy roadmap discussions to get everyone on the same page on how to optimize priorities globally.
This allows changes that are needed in central components to be implemented in an orderly fashion.
```
Split into multiple sentences.

File: product/02-innersource-and-agile.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+What's more often there are so many requests coming from teams that one central component team runs out of capacity to implement all of the requested changes.
```
Comment (rrrutledge) [inline product/02-innersource-and-agile.md:16]:
```suggestion
Often there are so many requests coming from teams that one central component team runs out of capacity to implement all of the requested changes.
```

Simplify lead-in.

File: product/02-innersource-and-agile.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+While in traditional organizations there are only two ways of making changes to dependencies:
+* Submit a feature request/ bug report and wait for the other team to prioritize that change and implement it.
+* Build a workaround to avoid the bug or locally provide the functionality needed.
+If none of those options is successful typically the issue is being escalated and decided at a higher hierarchical level-
```
Comment (rrrutledge) [inline product/02-innersource-and-agile.md:21]:
```suggestion

If none of those options is successful typically the issue is being escalated and decided at a higher hierarchical level-
```
Extra newline prevents this sentence from being included in the preceding bullet.

**Review by rrrutledge — COMMENTED** [review]
Some feedback.  I will have to work more on this review later.  Thank you!

File: product/01-introduction.md (hunk 5 above)
Comment (MaineC) [reply product/01-introduction.md:11]: Thanks for that hint!

File: product/04-negotiation.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+One difference with InnerSource to regular inter-team negotation are oportunities to think out of the box:
+Imagine a contributor Bob who needs a very complex change in the InnerSource project maintained by Alice.
+Bob is just beginning to understand the code-base and would have trouble understanding it on his own.
+In addition mentoring him through the process would take Alice a lot of time.
+However Alice has several high priority but easy to implement features on her backlog.
+What if Bob took some of those issue off of her backlog and implemented them - in return Alice has time to work on the change that Bob needs?
+For the sake of transperancy those agreements should be explained to both, the host team and the contributing team.
+Otherwise they will have a hard time understanding why Bob and Alice are not working on the changes that each of their customers needs.
+
+For another example imagin a host team that is working on a very popular InnerSource project.
```
Comment (fioddor) [inline product/04-negotiation.md:59]:
```suggestion
For another example, imagine a host team that is working on a very popular InnerSource project.
```

File: product/05-shared-ownership.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+While this will often result in the highest level of committment from contributors it also requires a high level of coordination crossing team boundaries.
+It also requires the highest level of transparency when making decisions about the project.
+
+
+To summarize each governance level needs a different approach to collaboration and coordination:
+* Increased sharing increases the need for communication and co-ordination,
+* Increased shared accountabilities can slow down decision making.
+
+
+## Over communication
```
Comment (fioddor) [inline product/05-shared-ownership.md:85]:
```suggestion
## Explicit pointers for contribution
```

**Review by fioddor — APPROVED** [review]
Apart from minor typos, the improvement is significant. I suggest merging and starting another iteration if needed.

File: product/02-innersource-and-agile.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+They are done when development is finished.
+In InnerSource code changes are shared very early in the process, sometimes when nothing more than a rough sketch is done.
+The goal is to seek early feedback and mentoring.
+This is particularly helpful for teams that are on diverse schedules and cannot find any time for pair programming.
+Often teams have the aspiration that nobody walks alone - in reality though this often isn't much more than an aspiration never achieved.
+In particular where contributions cross team boundaries.
+
+Focus on written communication: The goal with InnerSource is for the project to be transparent enough so that developers who are not part of the team can understand project decisions and follow alogthe process of software creation.
+As a result all communication needs to be in a place that everyone interested in the conversation can follow along: written, public, searchable and linkable.
+The goal is not to reduce disturbing others - the goal is to make all project conversations transparent.
```
Comment (jonico) [inline product/02-innersource-and-agile.md:106]:
```suggestion
The goal is not to reduce distractions to others - the goal is to make all project conversations transparent.
```

File: product/02-innersource-and-agile.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+
+Remember last time you insisted that your product be well tested, preferably with automated tests so deployments can happen frequently and without human intervention?
+This goal now helps with InnerSource as well: Contributions are much easier if contributors can check locally if their changes are safe.
+Tests also ensure that the host team remembers to keep the contributed functionality if they are reminded of the reason for it by a failing test.
+
+Remember last time you insisted on your team spending time to follow the goal of "leave code in a better shape than you found it"?
+That mindset will also help you when moving to an InnerSource model: It makes sure that quality and co-hesion of code remains high even when there are multiple contributions from different sources.
+
+
+## Common misunderstandings when coming from agile teams
```
Comment (jonico) [inline product/02-innersource-and-agile.md:83]: ### Additional common misunderstandings

* "My engineers will now work on whatever projects they like, so I cannot benefit from their work any more / plan with their capacity"
* "I do not need to take great care in protecting my infrastructure credentials used in my projects as we are all a big InnerSource family"
* "I do not have to adopt my conversation tone and can trust that anybody who has access to my project will always keep company secrets"

**Review by jonico — COMMENTED** [review]
✨ work @MaineC, looking forward to discuss on Monday 😊

**Comment by rrrutledge** [comment]
Thank you for this review, @truemped ❗  I keep meaning to get to this but have not done so, yet.  I am glad that others are chiming in.  Great work!

File: product/02-innersource-and-agile.md (hunk 13 above)
Comment (MaineC) [reply product/02-innersource-and-agile.md:83]: Those are really valuable to have. Thank you for sharing those!

File: product/02-innersource-and-agile.md [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+Also solving problems together, pairing with others or in person hackathons are valuable to find solutions quickly.
+The team needs to make sure though that all project relevant decisions are kept in channels that everyone has access to.
+That also may mean to post-pone important project decisions until everyone is back from vacation or waiting for another day or two if those working in another country are now on holiday.
+This is not only relevant for coding decisions, but also relates to general project mission, roadmap and direction.
+Without that information contributors will have a hard time understanding which contributions will have a good chance of getting accepted.
+
+### Impact of trust
+
+All discussions in InnerSource projects are visible to everyone in the company.
+Blaming people for their errors, ridiculing them for their mistakes, talking behind their backs about what they did wrong is a sure fire way to kill that trust and leads to the failure of that InnerSource project.
```
[the remark this answers is not in the document]
Comment (MaineC) [reply product/02-innersource-and-agile.md:123]: Good idea - thank you!

File: product/05-shared-ownership.md [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+
+"We are using pull requests to develop our component - so we are using InnerSource on a daily basis".
+While using pull requests and reviews is a crucial component, it is just the baseline for InnerSource projects.
+Just because two projects you depend on use pull requests on a daily basis does not mean that their openess to team-external contributions is the same.
+
+InnerSource comes with different best practices.
+In order to avoid confusion and frusttration for contributors it is important for host teams to define for their InnerSource project which governance model they want to adopt.
+Much like in Open Source these governance levels can differ substantially.
+
+In the InnerSource Commons we provide an InnerSource pattern that defines at least three governance levels:
```
[the remark this answers is not in the document]
Comment (MaineC) [reply product/05-shared-ownership.md:60]: We definitely should. Thanks for catching this.

File: product/05-shared-ownership.md [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+* Source code is visible to everyone - but the team has no time to mentor contributors.
```
[the remark this answers is not in the document]
Comment (MaineC) [reply product/05-shared-ownership.md:61]: Good point!

**Comment by MaineC** [comment]
Thank you @jonico and @truemped for your feedback on this. It is very much appreciated.

I intend to add some content wrt. the suggestions from @jonico - I'll look into simplifying and shortening the text after that.

**Comment by rrrutledge** [comment]
Hi @MaineC - I'm not sure how these conflicts came into place, but would you like to look at them and choose which text is right to keep?

**Comment by MaineC** [comment]
I did a trial run of the first and last chapter today.

One obvious misunderstanding when talking to product ppl: The term "project" (as in "Hadoop is an Open Source project" or "animal-tracker as an InnerSource project") is confusing to product people, because there, projects have a defined beginning but also a defined end date.

The alternative term "product" is tricky as well though: Open Source product implies quality criteria and support criteria that just don't make any sense, likely something similar is true for InnerSource.

Long story short: We should make it very clear, what we mean when using the term project at the very beginning.

**Comment by rrrutledge** [comment]
I’ve run into the project/product terminology difficulty as well.  Haven’t solved it, mostly because I have been dealing primarily with engineering people.

**Comment by MaineC** [comment]
One more (meta) learning of the trial run:

If you have an ISPO, this is one of the trainings that may make sense to do interactively and with participants working in groups to fill in some of the content themselves. 

Why:
* the type of knowledge that people come with can be more divergent than with engineers
* conversations can reveal anti-patterns in your organisation that people are unaware are anti-patterns
* there seemed to be more questions in particular contrasting to other project management approaches that people already know about that are easier to answer interactively: "Wait, that's similar to how XYZ does it, right?" "Yeah, similar, but different in the following subtle details..."


Would still be nice to have videos to point ppl to.

**Comment by rrrutledge** [comment]
Thanks for doing this, @MaineC .  Maybe over time we develop other flavors of articles whise content the presenter can pull up depending on audience response.

**Comment by MaineC** [comment]
Apart from specific delivery - what are the next steps for the this pull requests?

In the hope that I managed to address most of the feedback points raised:

@rrrutledge / @lenucksi should we get this merged so we can iterate on details afterwards in new pull requests or are there any blockers that you think we should change before merging?

**Comment by MaineC** [comment]
Thinking out loud:


https://github.com/InnerSourceCommons/InnerSourceLearningPath/pull/520/#discussion_r1064174356

https://github.com/InnerSourceCommons/InnerSourceLearningPath/pull/520/#discussion_r1064173572

those two likely need to find their way into the doc. We also need to add some paragraph explaining our understanding of project and why we don't call it product. Not sure if that should be done here or in additional PRs...

**Comment by rrrutledge** [comment]
@MaineC you have done some fantastic work and we are very much looking forward to it being merged! I can take a quick look this week and yes - let's see if it can go in 🎉 ❗

**Comment by rrrutledge** [comment]
Let's get this in to the _main_ branch now.  We can do any iteration on it there before putting it on the website.
